### PR TITLE
Eval transitions

### LINKED ISSUE #1765: [TASK]: Evaluate and change expressions in configuration matrix
We should be able to replace the expression with:
- another expression, or
- a literal numerical value

We should find a way to display both the 
- expression, and
- the numerical equivalent 

in the matrix cell if applicable

This task requires first-class symbolic support from the Pyodide/SymPy experiments

### PR DESCRIPTION
**Pull request by mj3cheun**
# Description

Screenshot 2023-08-31 at 12 48 07 AM
Screenshot 2023-08-31 at 12 47 50 AM

Resolves #1765

### CODE REVIEW

File: packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue [hunk 1]
```diff
@@ -73,13 +78,45 @@ const chosenRow = ref('');
...
+							}
+							output[cell.row][cell.col] = matrixVal;
+						}
+					})
+				)
+			)
+		);
+		matrixExpressionsList.value = output;
+	}
+);
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue:111]: Slight preference if we don't have multiple layers of promise alls, is it possible to do something like?

```
matrix.value.map(row => {
  row.map(cell => {
    matrixExpressionList.value[row][col] = await getMatrixValue(...)  
  })
})
```

File: packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue [hunk 2]
```diff
@@ -73,13 +78,45 @@ const chosenRow = ref('');
...
 const editableCellStates = ref<boolean[][]>([]);
 
+const matrixExpressionsList: any = ref([]);
+
+const parametersValueList = computed(() =>
+	props.modelConfiguration.configuration?.semantics.ode.parameters.reduce((acc, val) => {
+		acc[val.id] = val.value;
+		return acc;
+	}, {})
+);
```
[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue:88]: never mind this, got confused with something else.

File: packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue (hunk 1 above)
Comment (mj3cheun) [reply packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue:111]: fixed

File: packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue [hunk 3]
```diff
@@ -73,13 +78,45 @@ const chosenRow = ref('');
...
+		await Promise.all(
+			matrix.value.map(async (row) =>
+				Promise.all(
+					row.map(async (cell) => {
+						if (cell.value?.id) {
+							const matrixVal = await getMatrixValue(cell.value.id, props.shouldEval);
+							if (!output[cell.row]) {
+								output[cell.row] = [];
+							}
+							output[cell.row][cell.col] = matrixVal;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue:103]:
```suggestion
							if (!output[cell.row]) {
								output[cell.row] = [];
							}
							output[cell.row][cell.col] = await getMatrixValue(cell.value.id, props.shouldEval);
```

File: packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue [hunk 4]
```diff
@@ -118,6 +155,28 @@ function getMatrixValue(variableName: string) {
...
+async function getMatrixValue(variableName: string, shouldEvaluate: boolean) {
+	const expressionBase = getMatrixExpression(variableName);
+
+	if (shouldEvaluate) {
+		const expressionEval = (await pythonInstance.evaluateExpression(
+			expressionBase,
+			parametersValueList.value
+		)) as string;
+		return (
+			(await pythonInstance.parseExpression(expressionEval)) as {
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue]: Why not update create a `parseExpressionToMathML()` that return a `string` directly, so you don't have to do this weird typing `( .... as { mathml: string; }).mathml;`

File: packages/client/hmi-client/src/python/PyodideWorker.ts [hunk 5]
```diff
@@ -84,9 +84,16 @@ const parseExpression = (expr: string) => {
 	// function to convert expression to mathml
 	let result = pyodide.runPython(`
 		eq = sympy.S("${expr}", locals=_clash)
-		sympy.mathml(eq)
+		sympy.mathml(eq, printer="presentation")
```
Comment (YohannParis) [inline packages/client/hmi-client/src/python/PyodideWorker.ts:87]: Is this a parameters to choose between _presentation_ and _content_ MathML?
If yes, could you add a small comment like: 
```suggestion
		sympy.mathml(eq, printer="presentation") // printer refers to the type of MathML used "presentation" or "content"
```

Comment (mwdchang) [reply packages/client/hmi-client/src/python/PyodideWorker.ts:87]: oh it got changed to presentation mathML? That is going to have a problem when we save expressions, so I guess we need the function to return both.

File: packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue (hunk 4 above)
Comment (mj3cheun) [reply packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue]: this has been fixed by updating the `parseExpression` type

File: packages/client/hmi-client/src/python/PyodideWorker.ts (hunk 5 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/python/PyodideWorker.ts:88]: addressed this

**Review by mwdchang — APPROVED** [review]

**Pull request merged**